src/utils/attachment.ts

```diff
@@ -0,0 +1,42 @@
+import type { Attachment } from '../decorators/attachment/Attachment'
+import type { BaseName } from 'multibase'
+
+import { AriesFrameworkError } from '../error/AriesFrameworkError'
+
+import { BufferEncoder } from './BufferEncoder'
+import { HashlinkEncoder } from './HashlinkEncoder'
+import { JsonEncoder } from './JsonEncoder'
+
+/**
+ * Encodes an attachment based on the `data` property
+ *
+ * @param attachment The attachment that needs to be encoded
+ * @param hashAlgorithm The hashing algorithm that is going to be used
+ * @param baseName The base encoding name that is going to be used
+ * @returns A hashlink based on the attachment data
+ */
+export function encodeAttachment(
+  attachment: Attachment,
+  hashAlgorithm: 'sha2-256' = 'sha2-256',
+  baseName: BaseName = 'base58btc'
+) {
+  if (attachment.data.sha256) {
+    return `hl:${attachment.data.sha256}`
+  } else if (attachment.data.base64) {
+    return HashlinkEncoder.encode(BufferEncoder.fromBase64(attachment.data.base64), hashAlgorithm, baseName)
+  } else if (attachment.data.json) {
+    return HashlinkEncoder.encode(JsonEncoder.toBuffer(attachment.data.json), hashAlgorithm, baseName)
+  } else {
```

Comment:
Hashing JSON requires a canonicalization algorithm to consistently go from JSON to string/buffer. The `jws` property also doesn't support the `data.json` field at the moment. From the attachments RFC:

> Signatures over inlined JSON attachments are not currently defined as this depends upon a canonical serialization for the data.

We could use JSON Canonicalization Scheme (JCS) but that should also be added to the attachments RFC in that case.

I think for now we can just not support json for linked attachments?

Reply:
Yeah that sounds good. I did not think of that, I will remove it right away.